File: micro-common/src/main/java/com/jiangtj/micro/common/form/MobilePhoneHandler.java
package com.jiangtj.micro.common.form;

import com.jiangtj.micro.common.validation.MobilePhone;

import java.lang.reflect.Field;

public class MobilePhoneHandler implements FormRuleHandler<MobilePhone> {
    @Override
    public Class<MobilePhone> getAnnotation() {
        return MobilePhone.class;
    }

    @Override
    public FormRule handle(Field field, MobilePhone element) {
        Class<?> type = field.getType();
        if (type.isAssignableFrom(String.class)) {
            FormRule rule = new FormRule();
            rule.setType("string");
            rule.setPattern("^1[0-9]{10}$");
            rule.setMessage(element.message());
            return rule;
        }
        if (type.isAssignableFrom(Long.class)) {
            FormRule rule = new FormRule();
            rule.setType("number");
            rule.setMin(1000000000);
            rule.setMax(1999999999);
            rule.setMessage(element.message());
            return rule;
        }
        throw new RuntimeException("不支持的规则类型");
    }
}

File: micro-common/src/main/java/com/jiangtj/micro/common/form/MobilePhoneHandler.kt
package com.jiangtj.micro.common.form

import com.jiangtj.micro.common.validation.MobilePhone
import java.lang.reflect.Field

class MobilePhoneHandler : FormRuleHandler<MobilePhone> {
    override val annotation: Class<MobilePhone>
        get() = MobilePhone::class.java

    override fun handle(field: Field, element: MobilePhone): FormRule {
        val type = field.type
        if (type.isAssignableFrom(String::class.java)) {
            val rule = FormRule()
            rule.type = "string"
            rule.pattern = "^1[0-9]{10}$"
            rule.message = element.message
            return rule
        }
        if (type.isAssignableFrom(Long::class.java)) {
            val rule = FormRule()
            rule.type = "number"
            rule.min = 1000000000
            rule.max = 1999999999
            rule.message = element.message
            return rule
        }
        throw RuntimeException("不支持的规则类型")
    }
}
